lib/active_record_shards/model.rb

```diff
@@ -11,17 +11,22 @@ def not_sharded
     end
 
     def is_sharded? # rubocop:disable Naming/PredicateName
-      if self == ActiveRecord::Base
-        sharded != false && supports_sharding?
-      elsif self == base_class
-        if sharded.nil?
-          ActiveRecord::Base.is_sharded?
-        else
-          sharded != false
-        end
-      else
-        base_class.is_sharded?
-      end
+      # "sharded" here means self.sharded, but actually writing "self.sharded"
+      # doesn't work until Ruby 2.7 (and this gem currently supports 2.6).
```

Comment:
The following code works find in both Ruby 2.6.10 and 3.2.1. What did I miss?
```rb
module M
  attr_accessor :foo
  def bar
    self.foo
  end
end
class C
  extend M
end
C.bar
C.foo = 4
C.bar
# => 4
```

Reply:
oh. The important thing is that the `attr_accessor` is private heh. Forgot to mention that in the comment.

Reply:
I knew I missed something 😅